src/PageHeader/PageHeader.tsx

```diff
@@ -246,19 +315,19 @@ const LeadingAction: React.FC<React.PropsWithChildren<ChildrenPropTypes>> = ({
   sx = {},
   hidden = hiddenOnNarrow,
 }) => {
-  const {titleAreaHeight} = React.useContext(TitleAreaContext)
-
   return (
     <Box
+      data-component="PH_LeadingAction"
       sx={merge<BetterSystemStyleObject>(
         {
+          gridRow: GRID_ROW_ORDER.LeadingAction,
```

Comment:
I think we need to set a height to match the height of a single line of text. `align-items: center` aligns the leading and trailing actions to the vertical center of the text block, not the vertical center of the first line of text.

[Before:](https://primer.style/react/storybook/?path=/story/drafts-components-pageheader-features--with-leading-and-trailing-actions)
<img width="1766" alt="Screenshot 2023-12-13 at 6 20 25 PM" src="https://github.com/primer/react/assets/2313998/0fd444b8-8afa-45d5-b601-2e31b7398b51">

[After:](https://primer-402ed3ebc8-13348165.drafts.github.io/storybook/?path=/story/drafts-components-pageheader-features--with-leading-and-trailing-actions)
<img width="1766" alt="Screenshot 2023-12-13 at 6 19 41 PM" src="https://github.com/primer/react/assets/2313998/bbd9b550-948d-485b-bbb8-668a3c4b2c4f">

Reply:
I pushed a commit to fix this and used data-attributes as you recommended [here](https://github.com/primer/react/pull/3816#discussion_r1420804245). 

The challenge with this approach is the CSS specificity over sx styles. 

For example below, I tried to override the font-size of the title but it doesn't apply because the data-component selector has higher specificity. 

```jsx
<PageHeader.Title as="h1" sx={{fontSize: '14px'}}>
   PageHeader component initial layout explorations extra long pull request title
</PageHeader.Title>
```
<img width="1481" alt="pageheader title has 14px but the size isn't reflected and in the screenshot it shows the data attribute selector has more specificity than the generated sx class." src="https://github.com/primer/react/assets/1446503/c31ca7be-43e6-449b-b365-af4550705c4d">

We had the same problem with the button styles and used an [approach](https://github.com/primer/react/blob/71baff332b9da6911047d9e05cb83ee031b35a74/src/Button/Button.tsx#L68) to modify the sx styles selectors to have the same data attributes. How do you think about applying the same approach here or do you have a different idea? (You can see how that works in [the dev stories](https://primer.style/react/storybook/?path=/story/components-button-devonly--test-sx-prop))

Reply:
I think we can use the same approach as the buttons.

Reply:
Actually found out that our [other sx override approach](https://github.com/primer/react/blob/71baff332b9da6911047d9e05cb83ee031b35a74/src/Button/Button.tsx#L11) works better here 🙌🏻  Pushed a commit, seems like they can be overridden now ✨